src/Pate/Arch.hs

```diff
@@ -127,12 +143,80 @@
                 --
                 -- For example, these could be PLT stub symbols for ELF binaries
                 , validArchPatchedExtraSymbols :: Map.Map BS.ByteString (BVS.BV (MC.ArchAddrWidth arch))
+                , validArchStubOverrides :: ArchStubOverrides arch
                 }
 
+-- | A PLT stub is allowed to make arbitrary modifications to the symbolic state
+data StubOverride arch =
+  StubOverride
+    (forall sym v bin.
+      W4.IsSymExprBuilder sym =>
+      PB.KnownBinary bin =>
+      sym ->
+      PS.SimState sym arch v bin ->
+      IO (PS.SimState sym arch v bin))
+
+data ArchStubOverrides arch =
+  ArchStubOverrides (Map.Map BS.ByteString (StubOverride arch))
+
+lookupStubOverride ::
+  ValidArchData arch -> BS.ByteString -> Maybe (StubOverride arch)
+lookupStubOverride va nm = let ArchStubOverrides ov = validArchStubOverrides va in
+  Map.lookup nm ov
+
+mkMallocOverride ::
+  forall arch.
+  16 <= MC.ArchAddrWidth arch =>
+  MS.SymArchConstraints arch =>
+  MC.ArchReg arch (MT.BVType (MC.ArchAddrWidth arch)) {- ^ length argument (unused currently ) -} ->
+  MC.ArchReg arch (MT.BVType (MC.ArchAddrWidth arch)) {- ^ register for fresh pointer -} ->
+  StubOverride arch
+mkMallocOverride _rLen rOut = StubOverride $ \sym st -> do
+  let mr = PS.simMaxRegion st
+  let w = MC.memWidthNatRepr @(MC.ArchAddrWidth arch)
+  mr_nat <- W4.integerToNat sym (PS.unSE mr)
+  zero <- W4.bvLit sym w (BVS.mkBV w 2)
+  let fresh_ptr = PSR.ptrToEntry (CLM.LLVMPointer mr_nat zero)
+  mr_inc <- PS.forScopedExpr sym mr $ \sym' mr' -> do
+    one <- W4.intLit sym' 1
+    W4.intAdd sym' mr' one
+  return (st { PS.simMaxRegion = mr_inc, PS.simRegs = ((PS.simRegs st) & (MC.boundValue rOut) .~ fresh_ptr) })
```

Comment:
Can you add a comment that describes the semantics of the override, perhaps with a justification for the behavior?